[ansible] add cluster_public_hostname

talset opened the pull request:
The cluster_hostname is already present (to be able to reach the
private/internal vip kube api).
The goal of cluster_public_hostname is to have the same thing but for
public access to the api.

If the public vip domain is not part of the certificat, when you will
try to access to the api via kubectl you will have something like that :

Unable to connect to the server: x509: certificate is valid for
cluster_hostname, master0, master1 ..., not cluster_public_hostname

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/contrib/2452)
<!-- Reviewable:end -->

galexrt commented on ansible/roles/kubernetes/tasks/gen_certs.yml:
```diff
@@ -37,6 +37,7 @@
     HTTP_PROXY: "{{ http_proxy|default('') }}"
     HTTPS_PROXY: "{{ https_proxy|default('') }}"
     CLUSTER_HOSTNAME: "{{ master_cluster_hostname |default('') }}"
+    CLUSTER_PUBLIC_HOSTNAME: "{{ master_cluster_public_hostname |default('') }}"
```
Could you remove the spaces between `|default` and the variable.
(On line 39 and 40)

galexrt reviewed (changes requested):
Just the two spaces to remove. The other changes LGTM :+1:

talset commented:
I signed it!

talset commented:
I signed it!

talset commented:
I signed it!

ingvagabund commented:
@redhat-k8s-bot test please

talset committed:
ansible: add cluster_public_hostname

The cluster_hostname is already present (to be able to reach the
private/internal vip kube api).
The goal of cluster_public_hostname is to have the same thing but for
public access to the api.

If the public vip domain is not part of the certificat, when you will
try to access to the api via kubectl you will have something like that :

Unable to connect to the server: x509: certificate is valid for
cluster_hostname, master0, master1 ..., not cluster_public_hostname

talset commented:
Suggested changes have been applied

ingvagabund commented:
@redhat-k8s-bot test please

ingvagabund commented:
/lgtm
/approve
